tsdb/wal/wal.go

```diff
@@ -886,6 +910,25 @@ func (r *segmentBufReader) Read(b []byte) (n int, err error) {
 
 // Computing size of the WAL.
 // We do this by adding the sizes of all the files under the WAL dir.
-func (w *WAL) Size() (int64, error) {
+func (w *WALImplementation) Size() (int64, error) {
 	return fileutil.DirSize(w.Dir())
 }
+
+// NewNoopWAL returns a no-op WAL.
+func NewNoopWAL(dir string) WAL {
+	return &noopWAL{dir: dir}
```

Comment:
Do we ned dir here? Why not just using `""`

Reply:
So in https://github.com/prometheus/prometheus/pull/6679 (the mmap chunk PR), the directory for the chunks is taken from the WAL. In tests, that leads to a lot of dirs left behind if we don't pass a temp dir. Hence a dir here.